### DISCUSSION TITLE
jsr305 follow-up

### DISCUSSION THREAD

**Discussion by hazendaz** [spotbugs/spotbugs#1788] [General]
see https://github.com/eclipse-ee4j/common-annotations-api/pull/5 and follow all the related notes.  It appears this almost got picked up on separate PR but still thought to be used only by findbugs and spotbugs is not really referenced.  Since we still have this attached to spotbugs-annotations, I think someone here should comment to them as to why that should go into the common annotations.

New PR was this and now closed as it seemst they were trying to get ahold of really old information on jsr305 contributors.

https://github.com/eclipse-ee4j/common-annotations-api/pull/91

**Comment by KengoTODA** [comment]
I'm personally not so sure about what they want to do, but [`javax.*` package is prohibited to use by any third party libraries](https://eclipse-foundation.blog/2019/05/03/jakarta-ee-java-trademarks/) so their PR is invalid at this timing. It would be better to wait until they make the PR valid and ready for discussion.

**Comment by hazendaz** [comment]
I believe point was that the new PR is jakarta namespace.  The problem as I read it's too attached to find bugs and I think they need to see we are spot bugs now.  If I'm not mistaken spotbugs annotations is importing jsr 305.  So if we are still using it, wouldn't it make sense to go official under jakarta?

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by KengoTODA** [reply]
I see. Then I prefer waiting until the result comes; The annotation of `jakarta` package could be common, could be not so common. It's too early to make this judgement which gives impact on all spotbugs annotations users.

On the other hand, I think it could be nice to implement new detectors based on their annotation, to support their users. We can design it with keeping compatibility for existing detectors.

Note that there is [another activity making a new standard of annotation for static analysis](https://jspecify.dev/), which Eclipse does not attend. Better to watch both jakarta and jspecify, and support users of both if possible.

**Reply by gredler** [reply]
> Note that there is [another activity making a new standard of annotation for static analysis](https://jspecify.dev/), which Eclipse does not attend. Better to watch both jakarta and jspecify, and support users of both if possible.

JSpecify 1.0.0 has been released!

**Reply by JuditKnoll** [reply]
Issue about Supporting JSpecify: https://github.com/spotbugs/spotbugs/issues/3143
PR adding partial support of JSpecify: https://github.com/spotbugs/spotbugs/pull/3142

**Comment by hazendaz** [comment]
I think once jspecify is in, we should just drop jsr305, it causes far too many problems and its in violation of copyright ownership.  Oracle just hasn't threatened us yet and probably won't for historical reasons and we make no money but still, we need to get rid of it.  Its true a lot of people are stuck on legacy javax still but there are so many other solutions to get around it.
